### PR TITLE
Feature/mg misc

### PR DESCRIPTION
**Pull request by maddyscientist**
Quality of life improvement for multigrid to reduce unnecessary template instantiations and some Jitify fixes:
* Significantly reduced compilation time for multigrid-related kernels (setup kernels, copy, restrictor and block_orthogonalize kernels)
* Fix Jitify with CUB 1.8
* Increase `TuneKey::name_n` to prevent buffer overflow
* Add new `atomicAbsMax` function for floats to atomic.cuh which uses native integer atomic `atomicMax`
* Optimization of `compute_max` kernel for Yhat calculation to use `atomicAbsMax`
* Halved number of templates instantiated for Laplace and staggered operators (do not instantiate unnecessary dagger templates)

### CODE REVIEW

**Comment by maddyscientist** [comment]
Pushed a bunch more compile-time reductions for restrictor, block-ortho, Laplace and staggered dslash kernels.  This pull should be considered complete (barring review and Jenkins).

**Comment by maddyscientist** [comment]
@weinbe2 are you wanting to review this, or happy with @cpviolator's sign off?

**Comment by weinbe2** [comment]
@maddyscientist compiling and visual reviewing now

File: lib/coarsecoarse_op.cu [hunk 1]
```diff
@@ -3,19 +3,39 @@
...
 
 #define COARSECOARSE
-#ifdef GPU_MULTIGRID
 #include <coarse_op.cuh>
-#endif
+
 namespace quda {
 
-#ifdef GPU_MULTIGRID
+  static CloverField* dummyClover()
```
Comment (weinbe2) [inline lib/coarsecoarse_op.cu]: I assume this dummy field is just to enable using pre-compiled routines? If so (or if not), can you please add a comment noting the reason?

Comment (maddyscientist) [reply lib/coarsecoarse_op.cu]: Exactly that is the reason.

Comment (maddyscientist) [reply lib/coarsecoarse_op.cu]: Can add comment.

**Comment by weinbe2** [comment]
Pending my request for an extra comment, this looks good to me. I verified the code via a `ctest -V` run using 8 GPUs, as well as test of MG and staggered MG. Also did a quick single GPU test!

**Comment by maddyscientist** [comment]
@weinbe2 comment has been addressed.  I also fixed a few valgrind warnings I found when running a staggered MG test.

With this pull approved I'll merge this after Jenkins has finished (which should be a bit fast now at least with the improvements in this pull)/

**Pull request merged**